openpype/hosts/houdini/api/lib.py

```diff
@@ -989,22 +1001,30 @@ def self_publish():
     if result:
         return
 
-    current_node = hou.node(".")
-    inputs_paths = find_rop_input_dependencies(
-        current_node.inputDependencies()
-    )
-    inputs_paths.append(current_node.path())
+    if not nodes:
+        nodes = [hou.node(".")]
```

Comment:
Would this select all nodes in the scene? If so, the docstring should reflect this

Reply:
I don't really understand this default either.

Reply:
yes, it's intended to be `hou.pwd()`. 
I could think of two solution to eliminate any confusion. 
1. Comment, but I think this is not good, because no guarantees this script was called from inside a node. 
```suggestion
    if not nodes:
        #  get the current node if no nodes are passed
        nodes = [hou.pwd()]
```

2, update self publish button script, but I'm afraid it breaks the backward compatibility 
```
    button_parm = hou.ButtonParmTemplate(
        "ayon_publish_trigger",
        "{} Publish Trigger".format(label),
        script_callback="from openpype.hosts.houdini.api.lib import publish_trigger; "
                                  "publish_trigger(nodes=[hou.pwd()])",
        script_callback_language=hou.scriptLanguage.Python,
        join_with_next=True
    )
```

Reply:
I like 1. more, we used to do the same on many functions. It's the developer's fault if he adds that function in a place that's not inside a node